src/probot.ts

```diff
@@ -95,6 +95,9 @@ export class Probot {
 
     this.on = (eventNameOrNames, callback) => {
       if (eventNameOrNames === "*") {
+        this.log.warn(
+          `Using the "*" event with the regular Probot.on() function is deprecated. Please use the Probot.onAny() method instead`
```

Comment:
I think I'd use the `app` instance name, add a `[probot] ` prefix, and recommend to use `app.webhooks.onAny` instead of `app.onAny`, because in a future release (when we introduce OAuth APIs) we'll probably get rid of `app.on*` in favor of `app.webhooks.on*`, 

```suggestion
          `[probot] Using the "*" event with the regular app.on() function is deprecated. Please use the app.webhooks.onAny() method instead`
```

Reply:
Done